content/ros/ROS_Nodes_Topics_Messages/README.md

```diff
@@ -0,0 +1,280 @@
+# Practical part
+
+## Creating a ROS package
+
+- Navigate to workspace. Example:
+
+```
+cd ~/catkin_ws/src/
+``` 
+- Create a package named ```my_first_package```, with ```std_msgs``` and ```rospy``` as dependencies:
+```
+catkin_create_pkg my_first_package std_msgs rospy 
+```
+- Complie:
+```
+catkin build
+```
+
+<details><summary>More details:</summary>
+
+<br/>
+
+In order to let ROS commands find your scripts, launch files, source files, etc.., you have to create a package and put it inside the ```src``` folder of your workspace (ex. ```catkin_ws/src```).
+
+A package is nothing more than a folder with at least these two files:
+    
+    - package.xml
+    - CMakeLists.txt
+
+Normally, a package also includes Python scripts, C++ source files, launch files (we will see later), message definition files, service definition files, ...etc. It 
+is common for a ROS package to have the structure shown below:
+
+```
+package/
```

Comment:
In our code base, we usually follow this slightly different convention for ROS packages:

```
package
├── README.md
├── CMakeLists.txt
├── package.xml
└── ros
        ├── msg/
        ├── action/
        ├── srv/
        ├── scripts/
        ├── include/
        └── src/
```

This has the advantage that ROS-dependent and ROS-independent code can be separated more easily. In particular, if we (want to) have ROS-independent code in the package, we simply include a `common` directory in the package root directory and then put the code there.

Reply:
Ok I see.
But at this stage, they will probably go and read about ROS on their own, most tutorials assume this file structure. I think I can comment on that during the presentation. Once they get the idea it won't be a problem to change file structure? What do you think?

Also, the [minimal_ros_packages](https://github.com/mas-group/minimal_ros_packages) follow the conventional file structure as well, in my slides I also assume this file structure (like the figures and so..).

Reply:
Yeah, ok, that's reasonable.